lib/generators/hyrax/work/work_generator.rb

```diff
@@ -73,7 +44,7 @@ def register_work
     lastmatch = nil
     in_root do
       File.open(config).each_line do |line|
-        lastmatch = line if line =~ /^\s*config.register_curation_concern :.*\n$/
+        lastmatch = line if line =~ /config.register_curation_concern :(?!#{file_name})/
```

Comment:
I think we need to do away with prepending the `Note: order of registration...` comment unless lastmatch.  During revocation, the comment may or may not be there, depending on what versions of the generator have been invoked previously.